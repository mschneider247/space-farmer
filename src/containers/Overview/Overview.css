.red-text {
  color: red;
}

.overview-box {
  width: 30%;
  background-color: rgba(0, 0, 0, 80%);
  border: 1px solid green;
  margin-left: 10%;
  font-size: 1.2em;
}