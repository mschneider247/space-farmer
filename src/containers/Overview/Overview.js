import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import { connect } from 'react-redux';

export class Overview extends Component{
  render() {
    let currentProposal = this.props.proposals[this.props.proposals.length - 1]
    return (
      <>
        <h3>Overview</h3>
        <p>Proposal# {currentProposal.id}</p>
        <p>Supply Cost: {currentProposal.supplies.supplyCost}</p>
        <p>Supply Weight: {currentProposal.supplies.supplyWeight}</p>
        <p>Destination: {currentProposal.destination}</p>
        <p>Rocket Choice: {currentProposal.rocket.name}</p>
        <Link to='/proposals'><button>Try Again!</button></Link>
      </>
    )
  }
};

export const mapStateToProps = state => ({
  proposals: state.proposals
});

export default connect(mapStateToProps, null)(Overview);