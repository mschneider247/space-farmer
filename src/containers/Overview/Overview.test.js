import React from 'react';
import { shallow } from 'enzyme';
import { Overview} from './Overview';

describe('Overview Tests', () => {
    it('Overview should match its snapshot', () => {
      const proposals = [{key: 1, 
                          id: 1, 
                          rocket: {payloads : [{id: 1, lb: 100}, {id: 2}]}, 
                          destination : 1,
                          supplies: {supplyWeight : 100}
                        }];
      const wrapper = shallow(<Overview proposals={proposals}/>)
      expect(wrapper).toMatchSnapshot();
    });
})