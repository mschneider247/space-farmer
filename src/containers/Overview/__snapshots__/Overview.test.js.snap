// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Tests Overview should match its snapshot 1`] = `ShallowWrapper {}`;
